pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java

```diff
@@ -86,6 +86,13 @@ public BaseResources(MetadataStoreExtended store, TypeReference<T> typeRef, int
         return cache.get(path);
     }
 
+    public CompletableFuture<Optional<T>> getAsync(String path, boolean forceRefresh) {
+        if (forceRefresh) {
```

Comment:
It's this just defeating the purpose of using cache if we invalidate the entry before try to access it?
Theoretically not just partitioned topic but all other metadata can have this problem.
It seems the `MetadataCacheImpl` implements `Consumer<Notification>` which should properly invalidate the cache entry if path is  deleted, but it's somehow not added to `listeners` in `AbstractMetadataStore`, maybe fixing this can fix our problem?

Reply:
Yes I agree. But I think in this scenario, the final state of znode depends on the execution progress of both `MetadataCacheImpl#delete` and `MetadataCacheImpl#accept`. If the execution time of `accept` is shorter than that of `delete`, then the state of znode is `Optional.empty()` instead of `invalidated`. So I wonder if it should invalidate the path at the end of the delete operation, not replace with an `Optional.empty()`.